src/frontend/flight/sync.rs

```diff
@@ -368,6 +381,49 @@ impl SeafowlDataSyncManager {
 
         Ok(Arc::new(UnionExec::new(vec![input_plan, proj_plan])))
     }
+
+    // Based on the provided origins and sequence numbers deduce what is the current durable
+    // sequence number for each origin.
+    fn advance_durable(
+        &mut self,
+        url: String,
+        origin_sequence_numbers: Vec<(String, SequenceNumber)>,
+    ) {
+        for (origin, seq_num) in origin_sequence_numbers {
+            if let Some(origin_seqs) = self.seqs.get_mut(&origin) {
+                if let Some(seq) = origin_seqs.get_mut(&seq_num) {
+                    // Remove the pending location for this origin/sequence
+                    seq.locs.remove(&url);
+                }
+
+                let mut remove_count = 0;
+                for (seq_num, seq) in origin_seqs.into_iter() {
+                    if seq.locs.is_empty() && seq.last {
+                        // We've seen the last sync for this sequence, all pending locations
+                        // have been flushed to and there's no preceding sequence to be flushed,
+                        // so we're good to flag the sequence as durable
+                        self.origin_durable.insert(origin.clone(), *seq_num);
+
+                        remove_count += 1;
```

Comment:
Hm, in this example:

```
insert (table_1, lsn_1, last=true)
insert (table_2, lsn_2, last=true)
insert (table_1, lsn_3, last=true)
```

When we flush only table_1, what's preventing us from looking at `lsn_3`, seeing that it has no tables pending to flush (is that correct? do we add table_2 to this list for lsn_3 as well?) and advancing the durable LSN=3?

Reply:
Because we've exited the for loop for lsn 2 (in particular `seq.locs.is_empty()` is false in that case since the table 2 location is still pending).